tools/server/server-common.cpp

```diff
@@ -494,6 +494,18 @@ int32_t server_tokens::process_chunk(
     return 0;
 }
 
+server_tokens server_tokens::clone() const {
+    server_tokens res;
+    res.has_mtmd = has_mtmd;
+    res.tokens   = tokens;
+    for (auto it = map_idx_to_media.begin(); it != map_idx_to_media.end(); ++it) {
+        size_t idx = it->first;
+        const mtmd::input_chunk_ptr & chunk = it->second;
+        res.map_idx_to_media[idx] = mtmd::input_chunk_ptr(mtmd_input_chunk_copy(chunk.get()));
+    }
+    return res;
+}
+
```

Comment:
Now that we have this function, I think we can enable host-memory prompt caching with `mtmd`:

- Update this code:

https://github.com/ggml-org/llama.cpp/blob/6fb3226d0af625085def4bd3d74291d1ae3681ee/tools/server/server-task.cpp#L1396-L1404

- Remove this condition:

https://github.com/ggml-org/llama.cpp/blob/6fb3226d0af625085def4bd3d74291d1ae3681ee/tools/server/server-context.cpp#L886-L889

I haven't tested, but I think the only reason that prompt caching didn't work was because wasn't sure how to copy the `server_tokens`. So it's worth giving it a try after these changes.

Reply:
Yes it will be nice to enable RAM cache for mtmd. I created an issue so we can have a look later on: https://github.com/ggml-org/llama.cpp/issues/17821